integrations/huggingface-transformers/tutorials/sparsifying_bert_using_recipes.md

```diff
@@ -0,0 +1,150 @@
+  --report_to wandb
+```
+
+If the command runs successfully, you should have a model folder called `bert-base-12layers` in the provided model directory `MODELS_DIR`.
+
+## Applying Pruning Recipes
+
+Using the teacher model `bert-base-12layers` above, you can now train and prune a "student" BERT model on the same dataset using knowledge distillation. `SparseML` extends the training script `run_qa.py` with the following arguments to support recipes and knowledge distillation:
+
+- `--recipe`: path to a YAML recipe file that defines, among other information, the parameters and the desired sparsity levels to prune;
+- `--distill_teacher`: path to the teacher model for distillation; the student model is trained to learn from both its correct targets and those "instructed" by the teacher model;
+- `--distill_hardness`: ratio (in `[0.0, 1.0]`) of the loss defined on the teacher model targets;
+- `--distill_temperature`: the temperature used to soften the distribution of the targets.
+
+Additionally, you will use the argument `--onnx_export_path` to specify the destination folder for the exported ONNX model. The resulting exported model could then be used for inference with the `DeepSparse Engine`.
+
+The following command prunes the model in 30 epochs to 80% sparsity of the encoder layers:
+
+```bash
+python transformers/examples/pytorch/question-answering/run_qa.py \
+  --model_name_or_path bert-base-uncased \
+  --distill_teacher MODELS_DIR/bert-base-12layers \
+  --distill_hardness 1.0 \
+  --distill_temperature 2.0 \
+  --dataset_name squad \
+  --do_train \
+  --do_eval \
+  --evaluation_strategy epoch \
+  --per_device_train_batch_size 16 \
+  --learning_rate 5e-5 \
+  --max_seq_length 384 \
+  --doc_stride 128 \
+  --output_dir MODELS_DIR/bert-base-12layers_prune80 \
+  --cache_dir cache \
+  --preprocessing_num_workers 6 \
+  --fp16 \
+  --num_train_epochs 30 \
+  --recipe ../recipes/bert-base-12layers_prune80.md \
+  --onnx_export_path MODELS_DIR/bert-base-12layers_prune80/onnx \
+  --report_to wandb
+```
+
+The directory `recipes` contains information about recipes and training commands used to produce our BERT pruned models on the SQuAD dataset.
+
+### Dropping Layers
+
+In some situations, you might drop certain layers from a BERT model and retrain on your own dataset. `SparseML` supports these use cases with a modifier called `LayerPruningModifier` that can be used as part of a pruning recipe. As an example, below is a modifier that prunes layers 5th and 7th from a BERT model:
+```
+!LayerPruningModifier
+ layers: ['bert.encoder.layer.5', bert.encoder.layer.7']
+```
+
+The directory `recipes` contains recipes, for example `bert-base-6layers_prune80`, that drops six layers from a model before applying pruning.
+
+The following table presents the recipes in the directory, the corresponding results, and `wandb` logging for our pruned BERT models.
+
+| Recipe name | Description | EM / F1 | DeepSparse performance | Weight and Biases |
```

Comment:
This table's markdown didn't render for me when previewing the file through GitHub; may you double-check this, thanks!

Reply:
From my side it looks ok, probably issue with github?